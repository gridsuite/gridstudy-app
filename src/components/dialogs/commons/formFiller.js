/**
 * Copyright (c) 2023, RTE (http://www.rte-france.com)
 * This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0/.
 */

import React from 'react';
import Grid from '@mui/material/Grid';
import makeStyles from '@mui/styles/makeStyles';
import { Box } from '@mui/system';
import { useTheme } from '@mui/material/styles';

const useStyles = makeStyles((theme) => ({
    filler: {
        backgroundColor: theme.formFiller.background,
        display: 'flex',
        justifyContent: 'center',
        alignItems: 'center',
    },
}));

export const FormFiller = ({
    size = 12, // Default will take the whole row
    lineHeight, // Filler's height : by default 100%, or if specified, equal to X times a generic row in the form.
    children, // These elements will be centered in the filler
}) => {
    const classes = useStyles();
    const theme = useTheme();

    return (
        <Grid
            container
            spacing={2}
            sx={{ height: lineHeight ? undefined : 'calc(100% - 32px)' }}
        >
            <Grid
                item
                xs={size}
                align={'start'}
                sx={{
                    marginTop: theme.spacing(2),
                }}
            >
                <Box
                    className={classes.filler}
                    sx={{
                        height: lineHeight
                            ? theme.spacing(7 * lineHeight)
                            : '100%',
                    }}
                >
                    {children}
                </Box>
            </Grid>
        </Grid>
    );
};
