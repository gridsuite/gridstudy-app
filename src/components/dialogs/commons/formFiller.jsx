/**
 * Copyright (c) 2023, RTE (http://www.rte-france.com)
 * This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0/.
 */

import { mergeSx } from '@gridsuite/commons-ui';
import { Box, Grid } from '@mui/material';
import { useTheme } from '@mui/material/styles';

const styles = {
    filler: (theme) => ({
        backgroundColor: theme.formFiller.background,
        display: 'flex',
        justifyContent: 'center',
        alignItems: 'center',
    }),
};

export const FormFiller = ({
    size = 12, // Default will take the whole row
    lineHeight, // Filler's height : by default 100%, or if specified, equal to X times a generic row in the form.
    children, // These elements will be centered in the filler
}) => {
    const theme = useTheme();

    return (
        <Grid container spacing={2} sx={{ height: lineHeight ? undefined : 'calc(100% - 32px)' }}>
            <Grid
                item
                xs={size}
                align={'start'}
                sx={{
                    marginTop: theme.spacing(2),
                }}
            >
                <Box
                    sx={mergeSx(styles.filler, {
                        height: lineHeight ? theme.spacing(7 * lineHeight) : '100%',
                    })}
                >
                    {children}
                </Box>
            </Grid>
        </Grid>
    );
};
