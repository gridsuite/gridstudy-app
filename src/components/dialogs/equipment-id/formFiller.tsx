/**
 * Copyright (c) 2023, RTE (http://www.rte-france.com)
 * This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0/.
 */

import { mergeSx } from '@gridsuite/commons-ui';
import { Box, Grid } from '@mui/material';
import { Theme, useTheme } from '@mui/material/styles';
import { ReactNode } from 'react';

const styles = {
    filler: (theme: Theme) => ({
        display: 'flex',
        justifyContent: 'center',
        alignItems: 'center',
        backgroundColor: theme.palette.mode === 'light' ? 'rgba(0, 0, 0, 0.06)' : 'rgba(255, 255, 255, 0.09)', // from MUI code cf FilledInput.js
    }),
};

interface FormFillerProps {
    size?: number;
    lineHeight?: number; // Filler's height will be X times a generic row in the form (or 100% by default)
    children: ReactNode;
}
export function FormFiller({ size = 12, lineHeight, children }: Readonly<FormFillerProps>) {
    const theme = useTheme();

    return (
        <Grid container spacing={2} sx={{ height: lineHeight ? undefined : 'calc(100% - 32px)' }}>
            <Grid
                item
                xs={size}
                sx={{
                    marginTop: theme.spacing(2),
                    align: 'start',
                }}
            >
                <Box
                    sx={mergeSx(styles.filler, {
                        height: lineHeight ? theme.spacing(7 * lineHeight) : '100%',
                    })}
                >
                    {children}
                </Box>
            </Grid>
        </Grid>
    );
}
