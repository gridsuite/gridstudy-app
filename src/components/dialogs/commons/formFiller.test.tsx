import { RenderBuilder } from '@gridsuite/commons-ui';
import { createTheme, ThemeOptions } from '@mui/material';
import { FormFiller } from './formFiller';

const renderBuilder = new RenderBuilder().withTheme(
    createTheme({
        formFiller: {
            background: '#e6e6e6',
        },
    } as ThemeOptions)
);

describe('FormFiller', () => {
    it('should render the form filler component', () => {
        const { getByTestId } = renderBuilder.render(<FormFiller lineHeight={undefined}>test</FormFiller>);
        const formFiller = getByTestId('FormFiller');
        expect(formFiller).toBeDefined();
    });
});
