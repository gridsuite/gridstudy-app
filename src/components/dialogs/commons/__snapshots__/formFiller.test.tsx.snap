// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormFiller should render the form filler component 1`] = `
<div
  class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 css-iejm2v-MuiGrid-root"
  data-testid="FormFiller"
>
  <div
    align="start"
    class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1hzi63y-MuiGrid-root"
  >
    <div
      class="MuiBox-root css-1k4vu89"
    >
      test
    </div>
  </div>
</div>
`;
